packages/api/src/codegen/typescript.ts

```diff
@@ -0,0 +1,357 @@
+import type Oas from 'oas';
+import type { Operation } from 'oas';
+import type { HttpMethods, JSONSchema } from 'oas/@types/rmoas.types';
+import type { MethodDeclaration } from 'ts-morph';
+
+import CodeGenerator from '.';
+import { IndentationText, Project, QuoteKind } from 'ts-morph';
+import { compile } from 'json-schema-to-typescript';
+
+type OperationTypeHousing = {
+  types: {
+    params: Record<'body' | 'formData' | 'metadata', string>;
+    responses?: Record<string, string>;
+  };
+  operation: Operation;
+};
+
+export default class TSGenerator extends CodeGenerator {
+  userAgent: 'api/1.0.0';
+
+  project: Project;
+
+  types: Map<string, string>;
+
+  files: Record<string, string>;
+
+  constructor(spec: Oas) {
+    super(spec);
+
+    // @todo fill this user agent in with something contextual.
+    this.userAgent = 'api/1.0.0';
+
+    this.project = new Project({
+      manipulationSettings: {
+        indentationText: IndentationText.TwoSpaces,
+        quoteKind: QuoteKind.Single,
+      },
+      compilerOptions: {
```

Comment:
These compiler options will eventually be controlled via the forthcoming cli and abide by how folks want their autogen'd SDKs to target and output to.